test/unit/token-test.ts

```diff
@@ -41,6 +46,31 @@ describe("Bazaar Token", function () {
     it("Should have an initial total supply of 0", async function () {
       expect(await bazrToken.totalSupply()).to.equal(0);
     });
+
+    it("should allow vault to mint to depositor", async function () {
+      await vaultSigner.mint(depositor1.address, testAmount);
+      expect(await bazrToken.balanceOf(depositor1.address)).to.equal(testAmount)
+    })
+
+    it("should not allow wallet to send funds to non-vault address", async function () {
+      await vaultSigner.mint(depositor1.address, testAmount);
+      await expect(depositor1Signer.transfer(depositor2.address, testAmount)).to.be.revertedWith("Only allow to transfer to or from Vault");
```

Comment:
Minor nitpick

> Only allow to transfer to or from Vault

Consider changing the language to match that of the test name: 

> should not allow wallet to send funds to non-vault address

The test doesn't actually test that someone can't receive tokens from a non-vault address. But by inference, it's true that if a non-vault wallet can only send to the vault, then a non-vault wallet can never receive the tokens from another non-vault wallet.

Reply:
>But by inference, it's true that if a non-vault wallet can only send to the vault, then a non-vault wallet can never receive the tokens from another non-vault wallet

🧠